Metrics cleanup

fishy opened the pull request:
1. Change some of the references to statsd metrics in doc comments to prometheus metrics.
2. Remove the deprecated prometheus metrics/labels from the previous release.

While I'm here, also add go 1.20 to CI.

kylelemons commented:
Can you update the PR description with a list of all of the changed / removed metrics so it's easy to copy/paste into release notes?

fishy commented:
Removed metrics (all the replacements were already added from at least v0.9.12)
- `httpbp_server_panic_recover_total`: use `httpbp_server_recovered_panics_total` instead
- `kafkabp_consumer_rebalance_total`: use `kafkabp_consumer_rebalances_total` and `kafkabp_consumer_rebalance_failures_total` instead
- `thriftbp_server_panic_recover_total`: use `thriftbp_server_recovered_panics_total` instead
- `thriftbp_client_pool_exhausted_total`: use `thriftbp_client_pool_exhaustions_total` instead
- `thriftbp_client_pool_release_error_total`: use `thriftbp_client_pool_release_errors_total` instead
- `thriftbp_client_pool_active_connections`: use `thrift_client_pool_active_connections` instead
- `thriftbp_client_pool_allocated_clients`: use `thrift_client_pool_idle_connections` instead

Removed labels:
- `thrift_client_name` from `thrift_client_pool_*`: use `thrift_pool` instead.
